certbot/src/certbot/_internal/main.py

```diff
@@ -1226,18 +1239,24 @@ def enhance(config: configuration.NamespaceConfig,
     config.certname = cert_manager.get_certnames(
         config, "enhance", allow_multiple=False,
         custom_prompt=certname_question)[0]
-    cert_domains = cert_manager.domains_for_certname(config, config.certname)
-    if cert_domains is None:
+    cert_sans = cert_manager.sans_for_certname(config, config.certname)
+    if cert_sans is None:
         raise errors.Error("Could not find the list of domains for the given certificate name.")
+
+    # Silently ignore IP addresses from the certificate for now, since enhancements aren't
+    # supported for IP address certificates (requires changes to the plugin system).
+    cert_domains, _ = san.split(cert_sans)
```

Comment:
i don't think we should just silently ignore IP addresses here either. if the cert contains IPs, i think we should either:

1. error
2. use `logger.warning` to notify the user we're doing this